.github/workflows/ci-pipeline.yml

```diff
@@ -1,8 +1,14 @@
 name: Continuous Integration
-on: [pull_request]
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```

Comment:
This means you aren't running tests on your branch until you open a PR. I'd say that's too strict.